lib/phase/flatten.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const Hoek = require('hoek');
+const clone = require('clone');
+
+/**
+ * Flatten Phase
+ *  - Merges shared into jobs
+ *  - Converts complex environment into JSON strings
+ * @method
+ * @param  {Object}   parsedDoc Document that went through structural parsing
+ * @param  {Function} callback  Function to call when done (err, doc)
+ */
+module.exports = (parsedDoc, callback) => {
+    const doc = parsedDoc;
+
+    // Shared
+    // Flatten shared into jobs
+    Object.keys(doc.jobs).forEach((jobName) => {
```

Comment:
It might help with readability just to move both of these iterations into helper functions

i.e. having a `mergeJobsToShared` and a `cleanEnvironmentVariables` or something along those lines